testing/generate_test.py

```diff
@@ -0,0 +1,107 @@
+#
+# generate_test.py
+# Mich, 2016-06-16
+# This file is part of MLDB. Copyright 2016 Datacratic. All rights reserved.
+#
+import argparse
+import subprocess
+import datetime
+import os
+
+template = """#
+# {filename}
+# {author}, {date}
+# This file is part of MLDB. Copyright {year} Datacratic. All rights reserved.
+#
+
+mldb = mldb_wrapper.wrap(mldb)  # noqa
+
+class {classname}(MldbUnitTest):
+
+    @classmethod
+    def setUpClass(cls):
+        pass
+
+    def test_it(self):
+        pass
+
+if __name__ == '__main__':
+    mldb.run_tests()"""
+
+
+def get_filename():
+    parser = argparse.ArgumentParser(description=
+        "Create a test file skeleton and create the entry in the makefile.")
+    parser.add_argument('filename', nargs=1, help="The filename to create.")
+    args = parser.parse_args()
+    filename = args.filename[0]
+
+    if not filename.endswith('.py'):
+        filename += '.py'
+
+    if os.path.isfile(filename):
+        raise Exception("File already exists")
+
+    return filename
+
+
+def get_class_from_filename(filename):
+    capitalize_next = True
+    classname = ''
+
+    for c in filename:
+        if c == '.':
+            break
+        if c in ['_', '-']:
+            capitalize_next = True
+        else:
+            if capitalize_next:
+                classname += c.capitalize()
+                capitalize_next = False
+            else:
+                classname += c
+
+    return classname
+
+
+def update_makefile(filename):
+    for test in [('testing.mk', False), ('pro_testing.mk', True)]:
```

Comment:
for filename, pro in [('...', True), ...]
Ça évite d'avoir à faire des [0], [1] qui sont moins lisibles.